File: footspa-ios/footspa/footspa/Database/Entity/MoneyNodeType.swift
//
//  MoneyNodeType.swift
//  footspa
//
//  Created by ljx on 2025/11/13.
//

import Foundation
import GRDB

enum MoneyNodeType: String, Codable, DatabaseValueConvertible {
    case none = ""
    case `public` = "public"
    case outside = "outside"
    case third = "third"
    case employer = "employer"
    case employee = "employee"
    case customer = "customer"
    case card = "card"
}

File: footspa-android/app/src/main/java/com/spread/footspa/db/MoneyNodeType.kt
package com.spread.footspa.db

const val NODE_TYPE_EMPLOYER = "股东"
const val NODE_TYPE_EMPLOYEE = "员工"
const val NODE_TYPE_CUSTOMER = "顾客"
const val NODE_TYPE_THIRD = "三方平台"
const val NODE_TYPE_CARD = "卡"

enum class MoneyNodeType(val str: String) {

    None(""),
    Public("public"),
    Outside("outside"),
    Third("third"),
    Employer("employer"),
    Employee("employee"),
    Customer("customer"),
    Card("card")
}

val MoneyNodeType.displayStr: String
    get() = when (this) {
        MoneyNodeType.Employer -> NODE_TYPE_EMPLOYER
        MoneyNodeType.Employee -> NODE_TYPE_EMPLOYEE
        MoneyNodeType.Customer -> NODE_TYPE_CUSTOMER
        MoneyNodeType.Card -> NODE_TYPE_CARD
        MoneyNodeType.Third -> NODE_TYPE_THIRD
        else -> ""
    }